app/models/user_optional_field.rb

```diff
@@ -0,0 +1,13 @@
+class UserOptionalField < ApplicationRecord
+  belongs_to :user
+
+  validates :label, presence: true, length: { maximum: 30 }, uniqueness: true
```

Comment:
this makes it unique for the entire table, we want it to be unique for the user I guess :)
We should also add an underlying unique index in the migration, see [add_index](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/SchemaStatements.html#method-i-add_index)

Reply:
Added the scope to the validation.

I'm not very familiar with database indexes. You mean something like this?:
```ruby
add_index :user_optional_fields, :field , unique: true
```

Reply:
Okay apparently that didn't work, and neither did any of the other variations I tried in the course of this deep rabbit hole I fell into :upside_down_face: 

@rhymes Would appreciate some clarity (can be over a zoom call if you'd like).

Reply:
Never mind, the error I was seeing turned out to be unrelated.

I pushed what I **think** you were asking for, let me know if I made a mistake.